### PR TITLE
Adds hooks to createExpressServer

### PR DESCRIPTION
**Pull request by fastner**
Allows to plug in own middleware to server if e.g. the body must not be parsed before using the middleware

### CODE REVIEW

**Comment by swernerx** [comment]
I think it would be helpful to have a more detailed example.

**Comment by swernerx** [comment]
I wonder why not just pass an array of custom "middleware"... which is that looped through with `server.use(custom[xxx])`

**Comment by fastner** [comment]
My use case for this is by implementing an openid connect server with edgestack there is a generic middleware provided by https://github.com/panva/node-oidc-provider. This middleware has it's own secure interpreter for request bodies. In edgestack's createExpressServer() the bodyParser middleware is used. This prevents streaming libs to get body as stream.

This hook allows specific middleware to be attached to the middleware chain before any body/request parsing is done but after security middleware like helmet.

This can also be implemented by adding an array of custom middlewares to config, you're right. If you think this is the better way to do it here I'm fine in changing this PR in that way...

File: src/api/server/createExpressServer.js [hunk 1]
```diff
@@ -137,6 +137,16 @@ export default function createExpressServer(config)
   // @see https://helmetjs.github.io/docs/dont-sniff-mimetype/
   server.use(helmet.noSniff())
 
+  if (config.customMiddleware)
+    config.customMiddleware.forEach(
+      (middleware) => {
+        if (middleware instanceof Array)
```
Comment (fastner) [inline src/api/server/createExpressServer.js]: server.use has two signatures
- (middleware, [middleware, ...])
- (pathPrefix, middleware, [middleware, ...])

To support both a middleware and an array of string and middleware is allowed here

**Comment by swernerx** [comment]
Looks good.

**Pull request merged**